### PR TITLE
Update CoreFx packages  to beta-24217-04

### PR DESCRIPTION
**Pull request by ericstj**
/cc @weshaggard @stephentoub

### CODE REVIEW

**Comment by ericstj** [comment]
@dotnet-bot test this please

**Comment by ericstj** [comment]
@dotnet-bot test this please

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop Windows_NT Debug Build and Test

**Comment by ericstj** [comment]
I think I've weeded out the last of the issues here.

File: src/System.Security.Cryptography.X509Certificates/ref/project.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
     "System.IO": "4.0.0",
     "System.Runtime": "4.0.0",
     "System.Runtime.Handles": "4.0.0",
-    "System.Security.Cryptography.Primitives": "4.0.0-rc4-24215-01",
-    "System.Security.Cryptography.Algorithms": "4.2.0-rc4-24215-01",
-    "System.Security.Cryptography.Encoding": "4.0.0-rc4-24215-01"
+    "System.Security.Cryptography.Primitives": "4.0.1-beta-24217-04",
+    "System.Security.Cryptography.Algorithms": "4.2.1-beta-24217-04",
+    "System.Security.Cryptography.Encoding": "4.0.1-beta-24217-04"
```
Comment (bartonjs) [inline src/System.Security.Cryptography.X509Certificates/ref/project.json]: The src projects seem to be switching back to stable dependencies, and I suppose I can see an argument for the test projects requesting nearly-live dependencies; but I don't understand what position we seem to be taking with the ref dependencies

**Comment by bartonjs** [comment]
It seems like this is addressing what I was going for in issue #9418 (well, I was expecting test and ref project.json to switch to stable, too, to cut down on editing noise).

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

File: src/System.Security.Cryptography.X509Certificates/ref/project.json (hunk 1 above)
Comment (ericstj) [inline src/System.Security.Cryptography.X509Certificates/ref/project.json]: Ref needs to be cleaned up too.  Thanks for catching this.  I will fix it.

**Comment by ericstj** [comment]
All of the OSX release legs seem to failing in recent PRs with `[xUnit] [ERROR] - The plugin hasn't been performed correctly: Problem on deletion`.  This occurs after a completed build and test run with no errors during processing of test results.  @stephentoub @mmitche any idea what this is?

**Comment by ericstj** [comment]
Looks like @stephentoub opened https://github.com/dotnet/corefx/issues/9066 to track this.  Merging now since that's the only failure.

**Pull request merged**